spec/src/lib.rs

```diff
@@ -387,13 +393,44 @@ impl SystemCells {
         self.files.len()
     }
 
-    fn build_outputs(&self) -> Result<(Vec<packed::CellOutput>, Vec<Bytes>), Box<dyn Error>> {
+    fn build_outputs(
+        &self,
+        input: &packed::CellInput,
+        output_index_start: u64,
+    ) -> Result<(Vec<packed::CellOutput>, Vec<Bytes>), Box<dyn Error>> {
         let mut outputs = Vec::with_capacity(self.files.len());
         let mut outputs_data = Vec::with_capacity(self.files.len());
-        for res in &self.files {
+        for (offset, res) in self.files.iter().enumerate() {
             let data: Bytes = res.get()?.into_owned().into();
+            let data_hash = packed::CellOutput::calc_data_hash(&data).pack();
+
+            let type_script = if [
+                CODE_HASH_SECP256K1_BLAKE160_SIGHASH_ALL,
+                CODE_HASH_DAO,
+                CODE_HASH_SECP256K1_RIPEMD160_SHA256_SIGHASH_ALL,
+            ]
+            .iter()
+            .any(|hash| hash == &data_hash.unpack())
+            {
+                let output_index = output_index_start + offset as u64;
+                let mut blake2b = new_blake2b();
+                blake2b.update(input.as_slice());
+                blake2b.update(&output_index.to_le_bytes());
+                let mut ret = [0; 32];
+                blake2b.finalize(&mut ret);
+                let script_arg = Bytes::from(&ret[..]).pack();
+                let script = packed::Script::new_builder()
+                    .code_hash(TYPE_ID_CODE_HASH.pack())
+                    .hash_type(ScriptHashType::Type.pack())
+                    .args(vec![script_arg].pack())
+                    .build();
+                Some(script)
+            } else {
+                None
```

Comment:
Add panic to make sure all the hashes match perfectly?

Reply:
No, only 3 system cells can have type script, others remains empty.

Reply:
I see.